modules/tests/src/smithy4s/tests/PizzaClientSpec.scala

```diff
@@ -253,11 +258,15 @@ abstract class PizzaClientSpec extends IOSuite {
   }
 
   def router(ref: Compat.Ref[IO, State]) = {
-    def storeAndReturn(key: String, request: Request[IO]): IO[Response[IO]] = {
-      ref
-        .updateAndGet(_.saveRequest(key, request))
-        .flatMap(_.getResponse(key))
-    }
+    def storeAndReturn(key: String, request: Request[IO]): IO[Response[IO]] =
+      // Collecting the whole body eagerly to make sure we don't consume it after closing the connection
```

Comment:
Surprised we didn't run into this earlier - I was getting `java.io.IOException: response was disposed` when this was just keeping requests as-is.